### DISCUSSION TITLE
Can't figure out to upgrade all casks / packages at once

### DISCUSSION THREAD

**Discussion by dahanbn** [Homebrew/discussions#945] [Getting started] [answered]
I am new to a Mac and homebrew and somehow I have problems with updating. I regularly run `brew upgrade` and it upgrades some packages, e.g. zoom. Certain packages/casks, e.g. alfred or firefox won't get upgraded automatically. Even with `brew outdated` they don't show. Normally, I do not figure out that those casks / packages were outdated.

```
~ brew info --cask firefox
firefox: 85.0.2 (auto_updates)
https://www.mozilla.org/firefox/
/usr/local/Caskroom/firefox/81.0.2 (121B)
From: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/firefox.rb
==> Name
Mozilla Firefox
==> Description
Web browser
==> Languages
cs, de, en-CA, en-GB, en, eo, es-AR, es-CL, es-ES, fi, fr, gl, in, it, ja, ko, nl, pl, pt-BR, pt, ru, sv, tr, uk, zh-TW, zh
==> Artifacts
Firefox.app (App)
==> Analytics
install: 13,110 (30 days), 34,461 (90 days), 139,281 (365 days)
```

How can I achieve autoupdating for those packages? What do I miss? What would be the right command?

**Comment by SMillerDev** [comment, marked as answer]
If something auto updates won't update it for you unless you run `brew upgrade --greedy`

**Reply by dahanbn** [reply]
Thanks. That did it. It updates 16 packages right now. :-) I did not know that so many were outdated.

**Reply by SMillerDev** [reply]
This doesn't really mean that they were outdated. It just means that brew installed a version that was not the latest. Auto update would have kicked in for something like Firefox already.

**Reply by dahanbn** [reply]
I thought so but it never acted that way. 

```
==> Upgrading 16 outdated packages:
banking-4 7.4.2 -> 7.5.2,7650
firefox 85.0.2 -> 86.0
fontbase 2.13.2 -> 2.14.0
forklift 3.4.4 -> 3.4.4,211
glyphs 3.0.1,3038 -> 3.0.3,3066
google-chrome 87.0.4280.67 -> 88.0.4324.192,4324.192
iina 1.1.0 -> 1.2.0,129
iterm2 3.4.3 -> 3.4.4
macvim 8.2.1719,166 -> 169
obsidian 0.10.13 -> 0.11.0
omnidisksweeper 1.12.1 -> 1.13
raindropio 5.2.48 -> 5.2.49
rightfont 5.8.4 -> 5.9.0,2817
textexpander 6.5.5 -> 6.8.1,681.3
vlc 3.0.11.1 -> 3.0.12
webex-meetings latest -> latest
```

As you can see even security critical apps like Chrome and Firefox were not updated with a simple `brew upgrade` before.

**Reply by SMillerDev** [reply]
This doesn't take the actual app version into account. It's just comparing the version brew last installed with the latest version brew knows of. Firefox auto updating would not impact this list.
